src/main/java/net/mcreator/element/types/Item.java

```diff
@@ -48,11 +50,11 @@
 		implements IItem, IItemWithModel, ITabContainedElement, IItemWithTexture {
 
 	public int renderType;
-	public String texture;
+	@TextureReference(TextureType.ITEM) public String texture;
 	@Nonnull public String customModelName;
 
 	public Map<String, Procedure> customProperties;
-	public List<StateEntry> states;
+	@TextureReference(TextureType.ITEM) @ResourceReference("model") public List<StateEntry> states;

```

Comment:
Just for my understanding, if this marker is not present, the checker will not go checking list elements to check if they have reference tags?

Reply:
Correct. Unless type matches, these annotations are the only way to include field/method in lookup for performance reasons.